File: FacturacionWebAPI-master/Facturacion.Models/ItemVenta.cs
using System.ComponentModel.DataAnnotations;

namespace Facturacion.Models
{
    public class ItemVenta
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "El número de factura es requerido")]
        public int NumeroFactura { get; set; }

        [Required(ErrorMessage = "El producto es requerido")]
        public int ProductoId { get; set; }

        [Required(ErrorMessage = "El precio unitario es requerido")]
        public decimal PrecioUnitario { get; set; }

        [Required(ErrorMessage = "La cantidad es requerida")]
        public int CantidadVendida { get; set; }

        public virtual Factura? Factura { get; set; }

        public virtual Producto? Producto { get; set; }
    }
}

File: FacturacionWeb-master/src/app/api/models/item-venta.ts
/* tslint:disable */
/* eslint-disable */
import { Factura } from './factura';
import { Producto } from './producto';
export interface ItemVenta {
  cantidadVendida: number;
  factura?: Factura;
  id?: number;
  numeroFactura: number;
  precioUnitario: number;
  producto?: Producto;
  productoId: number;
}
